tests/test_hierarchy_helpers.hpp

```diff
@@ -50,11 +51,30 @@ class Source final : public dealii::Function<dim>
 };
 
 template <int dim>
-class ConstantMaterialProperty final : public dealii::Function<dim>
+class Coefficient : public dealii::Function<dim>
+{
+public:
+  virtual dealii::VectorizedArray<double>
+  value(dealii::Point<dim, dealii::VectorizedArray<double>> const &p,
+        unsigned int const = 0) const = 0;
+
+  virtual double value(dealii::Point<dim> const &p,
+                       unsigned int const component = 0) const override = 0;
```

Comment:
You don't really need this member function here, do you?

Reply:
I do for the matrix-based

Reply:
What I was trying to say is that the base class already defines this function as pure virtual and we could omit the definition here.
If you want to keep it for documentation purposes/helping to understand the code, it's fine with me to keep it, though.

Reply:
Oh I see. Yes, I think it's easier to see what is necessary to implement if everything is here. Otherwise, people who don't know `dealii::Function` won't know that they have to implement it.